app/static/ringing_room.js

```diff
@@ -2467,35 +2488,48 @@ $(document).ready(function () {
                         bell_circle.pull_rope_by_hand(LEFT_HAND);
                     }
 
-                    // Calls are: g = go; h = stop; b = bob; n = single.
                     if (["b", "B"].includes(key)) {
                         // console.log('calling bob');
-                        bell_circle.make_call("Bob");
+                        bell_circle.make_call("BOB");
```

Comment:
I believe changing the spelling of these calls will confuse Wheatley. Old versions just wouldn't ring, they'd never receive the string `Look to`.

Happy to push a new version that handles both spellings for an easy migration if you need to change this.
Is the other option to send the same string and do the mapping to find the audio instead of mapping to the display message?

Reply:
Yeah this will cause breakages, though perhaps it'd be a good reason to make people upgrade :shrug:.  If we're gonna break Wheatley then we should be OK with changing how `s_user_left` works in #236, since if it's already broken then breaking two things doesn't matter.

On an unrelated note, I had an idea actually that we should make CLI Wheatley actually be a separate package which runs `pip install --upgrade wheatley` and then runs the original code.  This way, everyone is always up to date, but I'm getting Windows forced update vibes which I don't like.

Reply:
The change in spellings had to do with using the mapping object — mostly, just trying to make sure there is an easy way to determine whether a call is "recognized". I'd prefer to keep the new spellings if possible.

Reply:
@kneasle whoops, had replied without refreshing so I didn't see your comment.

Ok, so it looks like I've planned somewhat badly: This change is due to go out tomorrow night, as the `sorry` call is part of an April 1 gag. (As is anticlockwise ringing, and something else, though everything except the something else is planned to be permanent; the something else is feature-flagged.) I hadn't thought through how this would affect Wheatley.

Unless one of you is already working on it, I'm gonna go put a PR up on Wheatley to hopefully address this.

Reply:
Ooh I wondered if the timing was significant :sweat_smile:.  When's the deadline for April Fools Day pranks?  I was thinking of making Wheatley rickroll people, but it might take a bit of effort to implement without actually breaking things.  I was thinking of covering the Wheatley box with a message that's like:
```
After this update, Wheatley has officially gained sentience
and wants to be thanked for his work.

[Thank Wheatley] <- this button rickrolls people
[... something else ...] <- this button restores the Wheatley box
```

I'm not 100% happy with this, but any suggestions for better wording are welcome.

Reply:
Ok so in that time frame, I think the key is that on `s_call` event the `call` string hasn't changed case, so all versions of Wheatley would still receive `Look to` and ring.

How about something like add a new `type` or `audio_type` field to `c_call` and `s_call` which can be the `BOB`, `ALL`, `SORRY` you changed to. 
That would also give options to send text (`call`) and which audio should be played independently (e.g. Silent, Change method, etc.)
Assuming it handled the lack of `type` being set on `c_call` and `s_call` as no audio, Wheatley v0.6's new composition calling feature (which wouldn't be sending `type`) would just be sending calls to be displayed, and we can add that in a patch to fix audio.

Reply:
We'll need to be able to launch by 3am UK time on April 1st (i.e. tomorrow evening US time). Our guiding principle when deciding on pranks has been "don't break anyone's workflow" — so personally I think requiring users to click something new to access existing Wheatley functionality is probably not great. I wonder about something more like having a rickroll option show up as the 2nd result in ever method search?

Reply:
(oops, same as before, I didn't see @centerboards last comment, but: I think the solution I have in place is fine, at least until we do a major code cleanup. Let me know if either of you think otherwise.)

Reply:
> We'll need to be able to launch by 3am UK time on April 1st

Ah OK I mean I think that I probably won't have time anyway, and tbh there are much easier and more effective ways of rickrolling people (clickbaity fb posts is the way to go for maximum rickrollage).